.github/workflows/conda-python-simulator.yaml

```diff
@@ -0,0 +1,107 @@
+on:
```

Comment:
This file looks to be very similar to `conda-python-build.yaml`

```
20c20
<         default: "ci/build_python.sh"
---
>         default: "ci/test_conda_simulator.sh"
27,29d26
<       upload_to_anaconda:
<         type: boolean
<         required: false
56c53
<       - name: Compute Build Matrix
---
>       - name: Compute Simulator Test Matrix
69c66
<   build:
---
>   test:
```

I wonder if some of this can be factored out to somewhere common, I'll fiddle with it a bit locally and see if I can provide a patch.

Reply:
Additionally - the simulator-test.yaml file is also very similar, but I don't expect it to remain that way. I feel like I need to get this PR over the line before making further modifications, though, as every small churn on `main` needs a corresponding update here. 

Since I hope to evolve the simulator testing, I'd rather not do / undo factoring things together with the build yaml files across two (or more PRs).